packages/core/src/impl/WriteApiImpl.ts

```diff
@@ -219,11 +221,18 @@ export default class WriteApiImpl implements WriteApi, PointSettings {
     })
     return this
   }
-  convertTime(value: string | undefined): string | undefined {
-    if (typeof value === 'string') {
-      return value ? value : undefined
-    } else {
+  convertTime(value: string | number | Date | undefined): string | undefined {
+    if (value === undefined) {
       return this.currentTime()
+    } else if (typeof value === 'string') {
+      return value.length > 0 ? value : undefined
+    } else if (value instanceof Date) {
+      return this.dateToProtocolTimestamp(value)
+    } else if (typeof value === 'number') {
+      return String(Math.floor(value))
```

Comment:
What about infinite value? 

`new Point('test').floatField('value', 2).timestamp(Math.pow(10, 1000))`

Reply:
Such values will fail on the server with an error being returned back. These errors indicate what is exactly wrong.